src/link/MachO/load_commands.zig

```diff
@@ -467,8 +467,34 @@ pub inline fn appleVersionToSemanticVersion(version: u32) std.SemanticVersion {
     };
 }
 
-pub fn inferSdkVersionFromSdkPath(path: []const u8) ?std.SemanticVersion {
-    const stem = std.fs.path.stem(path);
+fn readSdkVersionString(arena: Allocator, dir: []const u8) ![]const u8 {
+    const sdk_path = try std.fs.path.join(arena, &.{ dir, "SDKSettings.json" });
+    const contents = try std.fs.cwd().readFileAlloc(arena, sdk_path, std.math.maxInt(u16));
+    const parsed = try std.json.parseFromSlice(std.json.Value, arena, contents, .{});
+    if (parsed.value.object.get("MinimalDisplayName")) |ver| return ver.string;
+    return error.SdkVersionFailure;
+}
+
+pub fn inferSdkVersion(comp: *const Compilation) ?std.SemanticVersion {
+    var arena_allocator = std.heap.ArenaAllocator.init(comp.gpa);
```

Comment:
```suggestion
pub fn inferSdkVersion(gpa: Allocator, comp: *const Compilation) ?std.SemanticVersion {
    var arena_allocator = std.heap.ArenaAllocator.init(gpa);
```

I would prefer if we used an allocator provided by the driver aka the linker, even if they may be the same.

Reply:
done